docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
+1. As a user, I should be able to add a `Header` and `Footer` to PDF and PNG reports.
+    1. As a user, I should see Report title and a sample paragraph in the `header` by Default.
+    1. As a user, I should see date, time and report name in the `footer` by Default.
+1. As a user, I should be able to select which charts/visualizations I can include in a PDF or PNG report. (p2)
+
+##### Report trigger
+1. As a user, I should be able to select a report trigger type that belongs to one of two categories: `Schedule`, and `Alert`.
+1. As a user, when I select `Schedule`, I should be able to select a request time of now, future date, recurring, or custom cron expression (similar to alerting).
+    1. As a user, if I select `now`, I should be able to generate a report immediately after saving the report definition.
+    1. As a user, if I select `future date`, I should be able to configure the date and time the report will be generated.
+    1. As a user, if I select `recurring`, I should be able to configure a recurring schedule.
+    1. As a user, if I select `custom cron`, I should be able to configure a cron-based schedule.
+    1. As a user, if I select `recurring` or `custom cron`, I should be able to configure an end date.
+1. As a user, when I select `Alert`, I should be able to select the alert a report gets triggered from.
+
+##### Delivery settings
+1. As a user, I should be able to deliver a report to Kibana and/or Email recepients.
+1. As a user, when I set Kibana as a delivery channel, I should be able to select which Kibana users the report gets delivered to.
+1. As a user, when I set email as a delivery channel, I should be able to set recepients, email subject, and email body.
+1. As a user, I should be able to insert a file URL refecence to the email body.
+1. As a user, I should be able to insert the report source URL reference to the email body.
+1. As a user, I should be able to attach the report file to an email.
+1. As a user, I should be able to set the file format of report attachment (pdf, png | csv, excel, depending on the source).
+
+#### Report creation
+1. As a user, when I create a report definition, the definition settings are saved and show up as a list item in the ‘Report definitions’ list.
+1. As a user, when a report definition has been triggered via schedule or alert, I should be able to see an instance of the report in my "Reports" list, and any users I send the report to, will also see a report instance in their repective "Reports" list page.
+1. As a user, when a report is created from a triggered definition condition, I should receive a notification on all configured delivery channels (Email, chime, slack, etc)
+1. As a user, when a report is created from a triggered definition condition, I should receive a kibana toast alert letting me know that the report is available.
+1. As a user, when a report fails due to an error, I should receive a kibana toast alert letting me know that the report failed.
```

Comment:
Same question as above

Reply:
Let's discuss this one as well.

Reply:
* The sender of a shared report should receive a toast informing them there was an error
* The owner of a report definition should receive a toast informing them that there was an error when triggering the report generation
* The receiver of a report will only see a report in the reports list, with a status of error. They will not see a toast.

Reply:
Are these toasts coming only when the user is in the Reporting plugin? Is it following the same practice as the comment above?

Reply:
*When I am a sender, or a definition owner*
The toast should come wherever I am in the application.

*When I am a receiver*
No toasts should display whatsoever for scheduled or triggered events. I should only see the "New reports" indicator on the reporting icon on the sidebar.